### PR TITLE
Raise an error when saving dict with blank key 

### PR DESCRIPTION
**Pull request by BartoszPrusak**
Saving a dict() or a Namespace that has an entry under key `""` is not allowed in Neptune. This change adds an appropriate error message for such case.

Saving a dict in a run that has `""` as a key led to `object has no attribute 'items'` error.\
That's because saving `run["foo"] = {"bar": 123}` actually saves the item as `run["foo/bar"] = 123`. Therefore, `run["foo"] = {"": 0.2}` is actually saved as `run["foo/"] = 0.2`.\
Since Neptune interprets `"foo"` and `"foo/"` as the same key, this code is understood as writing a dict and another value type under the same key, which is illegal.

### CODE REVIEW

File: src/neptune/attributes/namespace.py [hunk 1]
```diff
@@ -42,6 +43,8 @@
 )
 from neptune.types.namespace import Namespace as NamespaceVal
 
+_logger = logging.getLogger(__name__)
```
Comment (Raalsky) [inline src/neptune/attributes/namespace.py:46]: Let's use `from neptune.internal.utils.logger import logger` instead.

File: src/neptune/attributes/namespace.py [hunk 2]
```diff
@@ -99,6 +102,14 @@ def assign(self, value: Union[NamespaceVal, dict, Mapping], *, wait: bool = Fals
             value = NamespaceVal(value)
 
         for k, v in value.value.items():
+            if len(parse_path(k)) == 0:
```
Comment (Raalsky) [inline src/neptune/attributes/namespace.py:105]: It's Python my dear:
```python
>>> if not []:
...     print("It's working!")
...
It's working!
```

File: src/neptune/attributes/namespace.py [hunk 3]
```diff
@@ -99,6 +102,14 @@ def assign(self, value: Union[NamespaceVal, dict, Mapping], *, wait: bool = Fals
...
+                _logger.warning(
+                    'Key "%s" can\'t be used in Namespaces and dicts stored in Neptune. Please use a non-empty key '
```
Comment (Raalsky) [inline src/neptune/attributes/namespace.py:107]: Let's use f-strings instead :wink:

File: tests/unit/neptune/new/test_stringify_unsupported.py [hunk 4]
```diff
@@ -302,6 +302,35 @@ def test_assign__dict(self, run):
 
         assert run["regular"].fetch() == run["stringified"].fetch()
 
+    def test_assign__drops_dict_entry_with_practically_empty_key(self, caplog, run):
+        with assert_unsupported_warning():
+            run["unsupported"] = {"": Obj(), "x": Obj()}
+        with assert_unsupported_warning():
+            run["unsupported"] = {"///": Obj(), "x": Obj()}
+        logs = caplog.get_records("call")
```
Comment (Raalsky) [inline tests/unit/neptune/new/test_stringify_unsupported.py:310]: This looks like a pretty good pattern (catching warning) -> Could we extract it to context-manager? Check `assert_no_warnings` as a reference.

File: tests/unit/neptune/new/test_stringify_unsupported.py [hunk 5]
```diff
@@ -302,6 +302,35 @@ def test_assign__dict(self, run):
 
         assert run["regular"].fetch() == run["stringified"].fetch()
 
+    def test_assign__drops_dict_entry_with_practically_empty_key(self, caplog, run):
```
Comment (Raalsky) [inline tests/unit/neptune/new/test_stringify_unsupported.py:305]: `stringify_unsupported` looks like doesn't have an impact here. Could we move this tests somewhere else?

File: src/neptune/attributes/namespace.py (hunk 2 above)
Comment (Raalsky) [inline src/neptune/attributes/namespace.py:105]: I'm not sure about this value verification here. Some of the types that we have (`Artifact`, `File`, and a few ones) does have a verification inside of a constructor (`__init__`). Could we move this directly to the constructor of `Namespace` as well? In addition, let's just filter out all the unwanted values and log a combined warning like `"", "/" could not be used as a name of an attribute`

Comment (BartoszPrusak) [reply src/neptune/attributes/namespace.py:105]: I know, but I was on the fence on how many people are ok with that syntax :|

Comment (Raalsky) [reply src/neptune/attributes/namespace.py:105]: That's quite common for Python. https://stackoverflow.com/a/53522

File: tests/unit/neptune/new/test_stringify_unsupported.py (hunk 5 above)
Comment (BartoszPrusak) [reply tests/unit/neptune/new/test_stringify_unsupported.py]: Done, moved to `test_handler.py`

File: src/neptune/attributes/namespace.py (hunk 2 above)
Comment (BartoszPrusak) [reply src/neptune/attributes/namespace.py]: Done and done

File: tests/unit/neptune/new/test_stringify_unsupported.py (hunk 4 above)
Comment (BartoszPrusak) [reply tests/unit/neptune/new/test_stringify_unsupported.py]: Done

File: src/neptune/attributes/namespace.py (hunk 3 above)
Comment (BartoszPrusak) [reply src/neptune/attributes/namespace.py]: Done

File: src/neptune/attributes/namespace.py (hunk 2 above)
Comment (BartoszPrusak) [reply src/neptune/attributes/namespace.py]: Done

File: src/neptune/attributes/namespace.py (hunk 1 above)
Comment (BartoszPrusak) [reply src/neptune/attributes/namespace.py]: Done

**Review by Raalsky — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,9 @@
 - Fixed GitPython `is_dirty` failing on Windows ([#1371](https://github.com/neptune-ai/neptune-client/pull/1371))
 - Fix SSL errors after forking process ([#1353](https://github.com/neptune-ai/neptune-client/pull/1353))
 - Fixed support of stringify value in series attributes with step ([#1373](https://github.com/neptune-ai/neptune-client/pull/1373))
+- `dict`s and `Namespace`s that are written to runs and contain an empty string "" key now produce a warning and drop
+  the entry with such key instead of raising an
```
Comment (normandy7) [inline CHANGELOG.md:8]:
```suggestion
  the entry with such a key instead of raising an
```

**Review by normandy7 — COMMENTED** [review]
Nitpick; looks OK otherwise.

File: CHANGELOG.md (hunk 6 above)
Comment (BartoszPrusak) [reply CHANGELOG.md:8]: Done

**Review by normandy7 — APPROVED** [review]

**Pull request merged**